### PR TITLE
Add new package: buddy

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

File: var/spack/repos/builtin/packages/buddy/package.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+from spack import *
+import platform
+
+
+class Buddy(AutotoolsPackage):
+    """A Binary Decision Diagram library."""
+
+    homepage = "https://sourceforge.net/projects/buddy/"
+    url      = "https://sourceforge.net/projects/buddy/files/buddy/BuDDy%202.4/buddy-2.4.tar.gz"
+    list_url = url
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/buddy/package.py]:
```suggestion
    list_url = "https://sourceforge.net/projects/buddy/files/buddy"
```

File: var/spack/repos/builtin/packages/buddy/package.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    list_depth = 1
+
+    version('2.4', sha256='d3df80a6a669d9ae408cb46012ff17bd33d855529d20f3a7e563d0d913358836')
+
+    depends_on('m4',       type='build')
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/buddy/package.py]: This tarball already comes with a configure script, so these dependencies probably aren't necessary.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/buddy/package.py (hunk 2 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/buddy/package.py]: fixed in https://github.com/spack/spack/pull/16166/commits/d1db50ac0ebf56cdc4df55067447a44c54499ac3

File: var/spack/repos/builtin/packages/buddy/package.py (hunk 1 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/buddy/package.py]: fixed in https://github.com/spack/spack/pull/16166/commits/d1db50ac0ebf56cdc4df55067447a44c54499ac3

**Review by adamjstewart — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**